syntax = "proto2";

package sisyphus;

import "model.proto";

message SetLedIntensityRequest {
  optional double intensity = 1;
}

message SetLedIntensityResponse {}

message GetLedIntensityRequest {}

message GetLedIntensityResponse {
  optional double intensity = 1;
}

message InsertPatternRequest {
  optional string pattern = 1;
  optional int32 position = 2;
}

message InsertPatternResponse {}

message DeletePatternRequest {
  optional int32 position = 1;
}

message DeletePatternResponse {
  optional string pattern = 1;
}

message GetPatternsRequest {}

message GetPatternsResponse {
  repeated string upcoming_patterns = 1;
  repeated string past_patterns = 2;
}

service SisyphusService {
  rpc SetLedIntensity (SetLedIntensityRequest) returns (SetLedIntensityResponse);
  rpc GetLedIntensity (GetLedIntensityRequest) returns (GetLedIntensityResponse);
  rpc InsertPattern (InsertPatternRequest) returns (InsertPatternResponse);
  rpc DeletePattern (DeletePatternRequest) returns (DeletePatternResponse);
  rpc GetPatterns (GetPatternsRequest) returns (GetPatternsResponse);
}
